barretenberg/cpp/src/barretenberg/vm/avm/trace/trace.cpp

```diff
@@ -3593,6 +3578,27 @@ std::vector<Row> AvmTraceBuilder::finalize(bool range_check_required)
     /**********************************************************************************************
      * ALU TRACE INCLUSION
      **********************************************************************************************/
+    // Finalize cmp gadget of the ALU trace
+    auto cmp_trace = alu_trace_builder.cmp_builder.finalize();
+    auto cmp_trace_size = cmp_trace.size() * 5;
+    if (main_trace_size < cmp_trace_size) {
+        main_trace_size = cmp_trace_size;
+        main_trace.resize(cmp_trace_size, {});
+    }
+    for (size_t i = 0; i < cmp_trace.size(); i++) {
+        std::array<Row, 5> row_chunks_arr = {
+            main_trace.at(5 * i),     main_trace.at(5 * i + 1), main_trace.at(5 * i + 2),
+            main_trace.at(5 * i + 3), main_trace.at(5 * i + 4),
+        };
+
+        alu_trace_builder.cmp_builder.merge_into(row_chunks_arr, cmp_trace.at(i));
+        // UGHHH I DONT KNOW ENOUGH CPP TO DO THIS NICELY
```

Comment:
At least, you could loop :-) 
for (size_t j = 0); j<5 ; j++) {
    main_trace.at(5 * i + j) = row_chunks_arr[j];
}
Same to initialize row_chunks_arr

Reply:
Potentially, you are looking for a "reference array/vector". They were introduced in bb, see:
ref_array.hpp
ref_vector.hpp